### ISSUE TITLE
[bug] Electron Finder Icon Mangled for some MacOS environments

### ISSUE THREAD

**Issue by nothingismagick** [quasarframework/app-extension-icon-genie#73]
I know why this is happening, and have confirmed that the issue is that we need to have a bitmap as the source for 16x16 and 16x16@2x

https://github.com/idesis-gmbh/png2icons/issues/6

So there are two approaches we can use:
1. Work with png2icons to fix the issue in the underlying lib
2. If mac is detected use iconutil

**Comment by nothingismagick** [comment]
The thing is that we should enable cross-platform construction of electron apps, even if they aren't planned for distribution by the app store.

**Comment by nothingismagick** [comment]
I don't know what this means, but we can supply pdf at 1x sizes???
https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/custom-icons/

**Comment by nothingismagick** [comment]
Here is the list of .icns resources
https://en.wikipedia.org/wiki/Apple_Icon_Image_format#Icon_types

**Comment by webnoob** [comment]
> I don't know what this means, but we can supply pdf at 1x sizes???
> https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/custom-icons/

It also says in that same paragraph:

> PNG format is recommended only for intricate designs that require effects like shading, textures, and highlights.

So I'm not sure the finder icon would apply. Worth a try though.

**Comment by nothingismagick** [comment]
No, I think PDF won't apply to icns

**Comment by nothingismagick** [comment]
The truly bizarre thing is that the finder thumbnail for the icns file itself DOES show up properly

![Screen Shot 2019-06-21 at 11 10 02](https://user-images.githubusercontent.com/35242872/59911917-3f602380-9415-11e9-9f5f-28e8280b50a2.png)

But as you can see, here is the mangled part:

![Screen Shot 2019-06-21 at 11 12 03](https://user-images.githubusercontent.com/35242872/59912023-77fffd00-9415-11e9-868e-8310cd30bcb6.png)

This application icon is OBVIOUSLY data being interpreted as a bitmap that is in png format.

**Comment by nothingismagick** [comment]
Got a workaround.
So after digging around, I was able to find a toolchain that successfully creates 

ic10, ic14, ic13, ic12, ic11 as non-premultiplied ARGB bitmaps

SIPS & osxiconutils => this is obviously MAC only, but the approach is interesting and something that could probably be ported.

https://github.com/sveinbjornt/osxiconutils/blob/8b5038b34e8507bfe8226d29ae213d077a3258d1/IconFamily/IconFamily.h#L41-L47

```
sips -s format icns linux-512x512.png --out icon.icns
seticon icon.icns Rename.app
```

**Comment by nothingismagick** [comment]
Official workaround until this icns issue is resolved:
```
      builder: {
        mac: {
          icon: './src-electron/icons/icon.png'
        }
      }
```

**Comment by nothingismagick** [comment]
For reference, this also works (but mac only):
https://github.com/jamf/icns-Creator/blob/master/icns_creator.sh

**Comment by nothingismagick** [comment]
If this is happening to you (eg. on MacOS 10.12), please get in touch and let us know ALL of the sordid details. I want to resolve this.

**Comment by mn4367** [comment]
png2icons v2.0.0 is out, it should fix this problem.

**Comment by nothingismagick** [comment]
@mn4367 - Thanks, I'll check it out Monday and get back to you.

**Comment by nothingismagick** [comment]
I couldn't wait. I just tested it - works perfectly. THANKYOU 

Fix will be available in Icon Genie 1.0.1

**Comment by mn4367** [comment]
Glad that it works for you! v2.0.0 was indeed a major overhaul, learned a lot.
